docs/sources/flow/reference/components/loki.relabel.md

```diff
@@ -0,0 +1,182 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/loki.relabel
+title: loki.relabel
+---
+
+# loki.relabel
+
+The `loki.relabel` component rewrites the label set of each log entry passed to
+its receiver by applying one or more relabeling `rule`s and forwards the
+results to the list of receivers in the component's arguments.
+
+<!-- TODO(@tpaschalis) Add note about loki.transform
+  To manipulate the log entry itself, you can look at the loki.transform
+  component which allows you to run one or more 'stages' on the log entries.
+ -->
+
+If no labels remain after the relabeling rules are applied, then the log entry
+are dropped.
+
+The most common use of `loki.relabel` is to filter log entries or standardize
+the label set that is passed to one or more downstream receivers. The `rule`
+blocks are applied to the label set of each log entry in order of their
+appearance in the configuration file.
+
+Multiple `loki.relabel` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.relabel "LABEL" {
+  forward_to = RECEIVER_LIST
+
+  rule {
+    ...
+  }
+
+  ...
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`forward_to` | `list(receiver)` | Where the log entries should be forwarded to, after relabeling takes place. | | yes
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.relabel`:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+rule | [rule][] | Relabeling rules to apply to received log entries. | no
+
+[rule]: #rule-block
+
+### rule block
+
+{{< docs/shared lookup="flow/reference/components/rule-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`receiver` | `receiver` | The input receiver where log lines are sent to be relabeled.
+
+## Component health
+
+`loki.relabel` is only reported as unhealthy if given an invalid configuration.
+In those cases, exported fields are kept at their last healthy values.
+
+## Debug information
+
+`loki.relabel` does not expose any component-specific debug information.
+
+## Debug metrics
+
+
+* `loki_relabel_entries_processed` (counter): Total number of log entries processed.
+* `loki_relabel_entries_written` (counter): Total number of log entries forwarded.
+* `loki_relabel_cache_misses` (counter): Total number of cache misses.
+* `loki_relabel_cache_hits` (counter): Total number of cache hits.
+* `loki_relabel_cache_size` (gauge): Total size of relabel cache.
+
+## Example
```

Comment:
This reads like more of a tutorial than an example to me. What do you think about making this briefer, and saving the detailed prose for a more fleshed out logging tutorial?

Reply:
That sounds like a good idea to me.

Reply:
Thanks for the feedback! I think that an end-to-end logging tutorial will be worth putting some time into.

For now, I've shortened the example to be like the rest of the components!